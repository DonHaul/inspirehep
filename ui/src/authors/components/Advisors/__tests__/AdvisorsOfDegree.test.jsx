import React from 'react';
import { render } from '@testing-library/react';
import { fromJS } from 'immutable';

import { MemoryRouter } from 'react-router-dom';
import AdvisorsOfDegree from '../AdvisorsOfDegree';

describe('AdvisorsOfDegree', () => {
  it('renders other advisors', () => {
    const advisors = fromJS([
      {
        name: 'Yoda',
      },
      {
        name: 'Another Dude',
        degree_type: 'other',
      },
    ]);
    const { asFragment } = render(
      <MemoryRouter>
        <AdvisorsOfDegree advisors={advisors} degreeType="other" />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders phd advisors', () => {
    const advisors = fromJS([
      {
        name: 'Yoda',
        degree_type: 'phd',
      },
      {
        name: 'Another Dude',
        degree_type: 'phd',
      },
    ]);
    const { asFragment } = render(
      <MemoryRouter>
        <AdvisorsOfDegree advisors={advisors} degreeType="phd" />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the master advisor', () => {
    const advisors = fromJS([
      {
        name: 'Yoda',
        degree_type: 'master',
      },
    ]);
    const { asFragment } = render(
      <MemoryRouter>
        <AdvisorsOfDegree advisors={advisors} degreeType="master" />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
