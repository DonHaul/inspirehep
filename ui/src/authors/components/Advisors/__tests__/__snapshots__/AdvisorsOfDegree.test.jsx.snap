// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvisorsOfDegree renders other advisors 1`] = `
<DocumentFragment>
  <div
    class="__InlineList__"
    data-testid="inline-data-list"
  >
    <span
      class=""
    >
      Advisors: 
    </span>
    <ul>
      <li>
        <a
          href="/authors?q=Yoda"
        >
          Yoda
        </a>
        , 
      </li>
      <li>
        <a
          href="/authors?q=Another%20Dude"
        >
          Another Dude
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`AdvisorsOfDegree renders phd advisors 1`] = `
<DocumentFragment>
  <div
    class="__InlineList__"
    data-testid="inline-data-list"
  >
    <span
      class=""
    >
      PhD Advisors: 
    </span>
    <ul>
      <li>
        <a
          href="/authors?q=Yoda"
        >
          Yoda
        </a>
        , 
      </li>
      <li>
        <a
          href="/authors?q=Another%20Dude"
        >
          Another Dude
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`AdvisorsOfDegree renders the master advisor 1`] = `
<DocumentFragment>
  <div
    class="__InlineList__"
    data-testid="inline-data-list"
  >
    <span
      class=""
    >
      Master Advisor: 
    </span>
    <ul>
      <li>
        <a
          href="/authors?q=Yoda"
        >
          Yoda
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
